pkg/api/mlflow/controller/artifact.go

```diff
@@ -43,16 +45,23 @@ func (c Controller) GetArtifact(ctx *fiber.Ctx) error {
 		return err
 	}
 
-	ctx.Set("Content-Type", "application/octet-stream")
+	filename := common.GetFilename(req.Path)
+	ctx.Set("Content-Type", common.GetContentType(filename))
+	ctx.Set("Content-Disposition", fmt.Sprintf("attachment; filename=%s", filename))
+	ctx.Set("X-Content-Type-Options", "nosniff")
 	ctx.Context().Response.SetBodyStreamWriter(func(w *bufio.Writer) {
 		defer artifact.Close()
 		bytesWritten, err := io.CopyBuffer(w, artifact, make([]byte, 4096))
 		if err != nil {
-			log.Errorf("error encountered in %s %s: error streaming artifact: %s", ctx.Method(), ctx.Path(), err)
+			log.Errorf(
+				"error encountered in %s %s: error streaming artifact: %s",
+				ctx.Method(),
+				ctx.Path(),
+				err,
+			)
 		}
 		log.Debugf("GetArtifact wrote bytes to output stream: %d", bytesWritten)
```

Comment:
please move this line _below_ the flush

Reply:
and please make it consistent with all of our other calls to `SetBodyStreamWriter`

Reply:
placed in `func() err` like the others, let me know if that's what you meant